src/imperative/imperative_utils.h

```diff
@@ -477,8 +477,18 @@ inline void PushFComputeEx(const FComputeEx& fn,
         // copying A to B may not happen, and will corrupt A's memory.
         InvalidateOutputs(outputs, req);
       }
+      // add for mkldnn OP + no mkldnn OP
+      const auto is_mkldnn = Op::GetAttr<bool>("TIsMKLDNN");
+      if (!is_mkldnn.get(attrs.op, false)) {
+        std::vector<NDArray> inputs_fallback;
+        CreateDefaultInputs(inputs, &inputs_fallback);
+        fn(attrs, opctx, inputs_fallback, req, outputs);
+      } else {
+#endif
+        fn(attrs, opctx, inputs, req, outputs);
```

Comment:
The macro condition check is a little bit complex here. Can we make it cleaner?

Reply:
OK.  Changed in [16652](https://github.com/apache/incubator-mxnet/pull/16652)